From 935f0550eef3c360899fc8bb84ba277907ebe299 Mon Sep 17 00:00:00 2001
From: Adam Cipoletti <adam@amcreativecoach.com>
Date: Fri, 18 Sep 2026 02:14:05 -0400
Subject: [PATCH 15/36] XSLT: accept libxml2 2.12's const error pointer

libxml2 2.12 made xmlStructuredErrorFunc take a const xmlError*, as
upstream WebKit adapted to in 2023.
---
 Source/WebCore/xml/XSLTProcessor.h          | 4 ++++
 Source/WebCore/xml/XSLTProcessorLibxslt.cpp | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/Source/WebCore/xml/XSLTProcessor.h b/Source/WebCore/xml/XSLTProcessor.h
index 0792f1ae..f5bc8694 100644
--- a/Source/WebCore/xml/XSLTProcessor.h
+++ b/Source/WebCore/xml/XSLTProcessor.h
@@ -62,7 +62,11 @@ public:
 
     void reset();
 
+#if LIBXML_VERSION >= 21200
+    static void parseErrorFunc(void* userData, const xmlError*);
+#else
     static void parseErrorFunc(void* userData, xmlError*);
+#endif
     static void genericErrorFunc(void* userData, const char* msg, ...);
     
     // Only for libXSLT callbacks
diff --git a/Source/WebCore/xml/XSLTProcessorLibxslt.cpp b/Source/WebCore/xml/XSLTProcessorLibxslt.cpp
index e125c604..cfca2411 100644
--- a/Source/WebCore/xml/XSLTProcessorLibxslt.cpp
+++ b/Source/WebCore/xml/XSLTProcessorLibxslt.cpp
@@ -81,7 +81,11 @@ void XSLTProcessor::genericErrorFunc(void*, const char*, ...)
     // It would be nice to do something with this error message.
 }
 
+#if LIBXML_VERSION >= 21200
+void XSLTProcessor::parseErrorFunc(void* userData, const xmlError* error)
+#else
 void XSLTProcessor::parseErrorFunc(void* userData, xmlError* error)
+#endif
 {
     PageConsoleClient* console = static_cast<PageConsoleClient*>(userData);
     if (!console)
-- 
2.50.1 (Apple Git-155)

